rules/Php74/Rector/Property/RestoreDefaultNullToNullableTypePropertyRector.php

```diff
@@ -133,4 +133,17 @@ private function isReadonly(Property $property): bool
         $tags = $phpDocInfo->getTagsByName('@readonly');
         return $tags !== [];
     }
+
+    private function isReadonlyClass(Class_ $class): bool
+    {
+        // native readonly
+        if ($class->isReadonly()) {
+            return true;
+        }
+
+        // @immutable annotation
+        $phpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($class);
+        $tags = $phpDocInfo->getTagsByName('@immutable');
```

Comment:
I am on mobile, `hasByName()` should actually can be used

https://github.com/rectorphp/rector-src/blob/f889e416d94cbfcf3734bb2fd4daf678aa11f8a9/src/BetterPhpDocParser/PhpDocInfo/PhpDocInfo.php#L188

Reply:
here you are: https://github.com/rectorphp/rector-src/pull/5797